Source/WebKit/NetworkProcess/storage/FileSystemStorageHandle.cpp

```diff
@@ -295,14 +298,21 @@ void FileSystemStorageHandle::requestNewCapacityForSyncAccessHandle(WebCore::Fil
     if (!isActiveSyncAccessHandle(accessHandleIdentifier))
         return completionHandler(std::nullopt);
 
-    if (newCapacity <= m_activeSyncAccessHandle->capacity)
-        return completionHandler(m_activeSyncAccessHandle->capacity);
+    uint64_t currentCapacity = m_activeSyncAccessHandle->capacity;
+    if (newCapacity <= currentCapacity)
+        return completionHandler(currentCapacity);
 
     if (!m_manager)
         return completionHandler(std::nullopt);
 
-    uint64_t spaceRequested = newCapacity - m_activeSyncAccessHandle->capacity;
-    m_manager->requestSpace(spaceRequested, [this, weakThis = WeakPtr { *this }, accessHandleIdentifier, newCapacity, completionHandler = WTFMove(completionHandler)](bool granted) mutable {
+    if (newCapacity < defaultInitialCapacity)
+        newCapacity = defaultInitialCapacity;
+    else if (newCapacity < defaultMaxCapacityForExponentialGrowth)
+        newCapacity = pow(2, (int)std::log2(newCapacity) + 1);
```

Comment:
Why not `newCapcity *= 2;`?

Reply:
The current heuristic tries not preallocating too much and not growing too slow.
`newCapcity *= 2` might allocate too much if the pattern is a big write followed by some small writes (e.g. 100MB -> 1MB -> 1MB). It might not differ too much if the pattern is continuous small writes (e.g. 1MB -> 1MB -> 1MB). 
We could check how it performs and make adjustment if needed.